crates/npmrc/src/custom_deserializer.rs

```diff
@@ -16,6 +16,18 @@ pub fn default_public_hoist_pattern() -> Vec<String> {
     vec!["*eslint*".to_string(), "*prettier*".to_string()]
 }
 
+pub fn is_home(path: String) -> bool {
+    path.starts_with("~/") || path.starts_with("~\\")
+}
+
+pub fn resolve_store_dir(pacquet_home: String) -> PathBuf {
+    if is_home(pacquet_home.clone()) {
+        let home_dir = dirs::home_dir().expect("Home directory is not available");
+        return home_dir.join(&pacquet_home[2..]);
+    }
+    PathBuf::from(pacquet_home.as_str())
```

Comment:
Instead of `is_home` and `&pacquet_home[2..]`, please use `pacquet_home.strip_prefix("~/")`.

Reply:
I see that you also strip `~\`, in which case, please use `pacquet_home.strip_prefix("~/").or_else(|| pacquet_home.strip_prefix("~\\"))`.

Reply:
https://github.com/pnpm/pacquet/pull/84/files#r1299147566

Reply:
Also, since you already computed `home` in `default_store_dir`, you should pass it as argument to this function instead of calling `dirs::home_dir()` again.